### DISCUSSION TITLE
Welcome to Acornima!

### DISCUSSION THREAD

**Discussion by adams85** [adams85/acornima#1] [Announcements]
## Welcome!

@lahma, as the main maintainer of Jint and Esprima, I figured this newest project of mine may be of some interest to you. So I decided to show it to you first, before making it public. I'm very curious to hear your opinion.

Also, I borrowed some of your work that you did in Esprima (string matcher sourcegen, test262 runner, etc.). Even though a lot of that code was heavily refactored/improved, I'd like to ask if you are ok with this. Of course, I was going to "credit" you in some way, I'd just like to know if you have any formal requirement to do so.

**Comment by lahma** [comment]
Awesome work @adams85 ! I've sometimes played with the acorn porting idea, but due to huge work that would be, dismissed the idea. But you did it! I see that the structures are quite familiar so shouldn't have such steep learning curve. Acorn was definitely a great choice to port from - times were different when JS Esprima was the the only big player... You just had to do this just after Jint 3.0 was released with hard Esprima dependency 😉

It will be a nice exercise at some point to check how well Jint would take in a different parser implementation, but probably not in the near future as the exhausting 3.0 work has finally come to an end. 

I have no claims for copyrights, credits etc. Most of of the original work was done by Sebastien and then the other contributors - where I think you have done quite the workload too! Probably best check Sebastien for licensing, BSD-3 and MIT are both very permissive licenses, of course MIT probably is a bit better recognized as such. I'm not a lawyer...

**Reply by adams85** [reply]
Thanks for the feedback and the kind words!

Yep, it took a good chunk of time (and it still has quite a long way to go), however Acorn looks like a safer bet for the future. Make no mistake, Esprima is pretty mature at this point and it does like 95% of the job well. But that remaining 5% seems pretty hard to figure out... It seemed just easier to go with an implementation that gets it 100% right in the first place, plus looks to have the steam necessary to keep this up for the upcoming JS versions.

Anyway, sorry for not being able to ship this before the major Jint release but at least we'll have something left for v4.0... 😄 BTW, the parser API and the AST is nearly identical to Esprima, so it shouldn't be a tough job to build an Acorn-based Jint. If you are interested in such an experiment, I'll be glad to help with that after pushing this one to v1.0.

Also, thank you for allowing to reuse your work and the advice about licensing too. I'll contact Sebastien regarding this soon and go public if I get the green light.

**Comment by lahma** [comment]
I could create a PR for a basic [NUKE build](https://nuke.build/) setup if you wish (including GitHub Actions), if you'd like?

**Reply by adams85** [reply]
I'm not familiar with NUKE but it looks like a solid tool. If it can save us from writing some YAML, that's a big win for sure. 😄 So I'd say, let's give it a go!
